File: android/src/main/java/com/lightningkite/khrysalis/observables/binding/ImageView.binding.actual.kt
package com.lightningkite.khrysalis.observables.binding

import android.graphics.BitmapFactory
import android.provider.MediaStore
import android.widget.ImageView
import com.lightningkite.khrysalis.net.HttpClient
import com.lightningkite.khrysalis.*
import com.lightningkite.khrysalis.observables.*
import com.lightningkite.khrysalis.rx.removed
import com.lightningkite.khrysalis.rx.until
import com.lightningkite.khrysalis.views.loadImage
import com.lightningkite.khrysalis.views.loadVideoThumbnail

/**
 *
 * Binds the imageview internal image to the image provided by the observable.
 * Any changes to the observable will cause a reload of the image to match the change.
 * An image can be from multiple sources, such as the web, an android image reference,
 * and a direct bitmap. It will handle all cases and load the image.
 *
 */
fun ImageView.bindImage(image: ObservableProperty<Image?>) {
    image.subscribeBy { it ->
        post {
            this.loadImage(it)
        }
    }.until(this.removed)
}

fun ImageView.bindVideoThumbnail(video: ObservableProperty<Video?>) {
    video.subscribeBy {
        post {
            this.loadVideoThumbnail(it)
        }
    }.until(removed)
}

File: ios/Khrysalis/src/observables/binding/ImageView.binding.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation
import AlamofireImage
import Alamofire


//--- ImageView.loadImage(Image?)

//--- ImageView.bindImage(ObservableProperty<Image?>)
public extension UIImageView {
    func bindImage(_ image: ObservableProperty<Image?>) -> Void {
        image.subscribeBy { it in
            self.loadImage(it)
        }.until(self.removed)
    }
    func bindImage(image: ObservableProperty<Image?>) -> Void {
        return bindImage(image)
    }
    
    func bindImage(_ image: ObservableProperty<Image>) -> Void {
        image.subscribeBy { it in
            self.loadImage(it)
        }.until(self.removed)
    }
    func bindImage(image: ObservableProperty<Image>) -> Void {
        return bindImage(image)
    }
    
    func bindVideoThumbnail(video: ObservableProperty<Video?>) -> Void {
        video.subscribeBy { it in
            self.loadVideoThumbnail(video: it)
        }.until(self.removed)
    }
}
